src/parse/errors.rs

```diff
@@ -440,604 +436,415 @@ impl Error {
         safe_exit(0)
     }
 
-    #[allow(unused)] // requested by @pksunkara
-    pub(crate) fn group_conflict<O, U>(
-        group: &ArgGroup,
-        other: Option<O>,
-        usage: U,
-        color: ColorChoice,
-    ) -> io::Result<Self>
-    where
-        O: Into<String>,
-        U: Display,
-    {
-        let mut v = vec![group.name.to_owned()];
-        let mut c = Colorizer::new(true, color);
-
-        start_error(&mut c, "The argument '")?;
-        c.warning(group.name)?;
-        c.none("' cannot be used with ")?;
-
-        let cause = match other {
-            Some(name) => {
-                let n = name.into();
-                v.push(n.clone());
-
-                c.none("'")?;
-                c.warning(&*n)?;
-                c.none("'")?;
-
-                format!("The argument '{}' cannot be used with '{}'", group.name, n)
-            }
-            None => {
-                let n = "one or more of the other specified arguments";
-
-                c.none(n)?;
-
-                format!("The argument '{}' cannot be used with {}", group.name, n)
-            }
-        };
-
-        put_usage(&mut c, usage)?;
-        try_help(&mut c)?;
-
-        Ok(Error {
-            cause,
-            message: c,
-            kind: ErrorKind::ArgumentConflict,
-            info: Some(v),
-        })
-    }
-
-    pub(crate) fn argument_conflict<O, U>(
+    pub(crate) fn argument_conflict(
         arg: &Arg,
-        other: Option<O>,
-        usage: U,
+        other: Option<String>,
+        usage: String,
```

Comment:
Why is this not `&str`?

Reply:
Because what we need here is `String`, not `&str`. This way the caller can decide how to form the string and reuse the already formed `String` without reallocation on our part. In fact, the caller has `String` most of the time.

Reply:
I was asking because the `usage` parameter in all the other errors below has `&str` but this one was different. I just wanted to confirm that it was intentional since it's not consistent.

Reply:
Yes, I was going to make the other methods to use `String` too (I have some ongoing work done locally), but decided to postpone it as quite irrelevant (PR is already larger than necessary). Do you want it in this PR or as a followup?

Reply:
If it's only `&str` to String and such minor stuff, you can add a new commit to this PR. We will squash after I review and approve the PR.